### DISCUSSION TITLE
A version for iOS/iPadOS 15 now possible?

### DISCUSSION THREAD

**Discussion by retrogamer0815** [FrostCo/AdvancedProfanityFilter#310] [Ideas]
Hi,

I love this extension and use it on all my desktop computers, unfortunately until now there just was never a way to get this kind of functuality within a browser under iOS or iPadOS. However now extensions for Safari under iOS and iPadOS 15 were announced and in WWDC video the developer demoed an extension which replaced the word "fish" on a website with fish emojis, which is kind of what AdvancedProfanityFilter does.

So any chance this extension could be made available for Safari on iPhone/iPad later this year when the new OS versions are released?

Here the video where they talk about and demonstrate extensions for Safari on iPhone and iPad:

https://developer.apple.com/videos/play/wwdc2021/10104/

Thanks :-)

**Comment by richardfrost** [comment]
I'm glad that you like the extension! I do think it should be possible with these latest update. I will definitely try to keep tabs on it as it becomes available and try to port it over.

Since I don't have a Mac (might be able to borrow one) or a developer account with Apple, I'll have to look into managing that cost, but this is a really exciting development. I know there are lots of iPhone/iPad users out there would definitely be excited about this. Thanks for the heads up!

**Reply by retrogamer0815** [reply]
Hopefully this extension can really be used the same way as on the desktop without too many limitations, it is not unusual for Apple to offer some new functions on iOS with some annoying limitations first which might be dealt with a year later ;-)

For example they talk about extensions being white listed website by website but I hope there will be a way to allow an extension to just work on any website without having to allow access every time you visit a new site.

Also in a second video about extensions they talk about "non-persistent background page support" which is something different on iOS compared to Safari on a Mac. I hope this doesn't mean that extensions have be manually turned on all the time, would love to hear your opinions and maybe later some updates about this when you had a chance to look into this a little further :-)

https://developer.apple.com/videos/play/wwdc2021/10027

Glad you accept donations, hope I can help a little with the development :-)

**Reply by richardfrost** [reply]
Yeah, it will be interesting to see how they implement it. Unfortunately they don't seem to have many details on how it will be implemented yet, so I guess we'll just have to wait and see. Extensions/Add-ons are going through a transition period right now for other browsers as well regarding security/privacy. For a while with manifest V3 (Google Chrome extensions), it sounded like they were really going to limit what extensions could do, but as things have evolved it looks like we'll be able to update APF without losing much, if any, functionality.

So, for now I'll try to keep tabs on how it develops, and then when they release it I'll do what I can to test it out. Thanks again for the info! Since I don't personally have an Apple device I don't keep tabs on them as much, but my wife does have an iPad I can test it on. Anyway, I'm really excited that it might soon be possible to make this filter available to Apple users in Safari.

**Reply by retrogamer0815** [reply]
Hi, not that I am in a hurry, I just read that public betas for iOS and iPadOS are now available. I am a little hesitant to install it on my phone since I rely it but it'd be ok for me to install it on my iPad in case you need some people for testing :-)

**Reply by richardfrost** [reply]
I haven't had a chance track anything down yet. I'll try this weekend to see if I can secure a macbook or something to take a look at it. I've got a few other things going too, so if I don't get to it this weekend, I should have some time to spend on it next week. Thanks again for staying current on this.

**Reply by richardfrost** [reply]
Okay, I decided it wasn't worth the hassle of trying to borrow a device, so I ended up ordering the cheapest mac I could find and will use that once it arrives. I'll let you know when I have something to test.

**Reply by retrogamer0815** [reply]
Thanks for going that far as to buy a new computer for this and no problem, once you have something to test, just let me know and I will install a public beta of iOS 15 on my iPad.

**Comment by retrogamer0815** [comment]
Hi, just want to let you know, that now both my phone and the iPad are running in iOS 15/iPadOS 15, since the final versions are available, so I can install Safari extensions now without having to install beta-versions of iOS or iPadOS :-)
I noticed by the way that some extensions for Safari which I know from Firefox or Chrome before like Dark Reader are not free but sold for a small amount in the AppStore. I think that is totally fine and fair, since it seems you have to do some more investments when developing for that platform, so I would gladly pay you for your work :-)

**Reply by richardfrost** [reply]
Thank you for your help! I have been successful in testing on a simulator, so hopefully it isn't far off now. I've applied to become an Apple developer, but haven't heard back yet from them. They said it could take up to 5 business days. Once that's done I'll see about getting it up to the store.

I also appreciate your willingness to pay for the app as well. I do think I'll set a modest price on it, because it costs $99 per year to be an Apple developer. I will definitely let you know as soon as its ready for testing! Thanks again.

**Reply by retrogamer0815** [reply]
Sounds great, thanks a lot for the update, looking forward to test this extension and finally have it on all my devices :-)

**Reply by richardfrost** [reply]
Sorry its taken so long @Joghurtbecher08115, but I finally have a test build if you're still interested in taking a look. Please fill out the form here: https://forms.gle/nwfZRSTzXpWa5vsH6 to get access.

**Reply by retrogamer0815** [reply]
> Sorry its taken so long @Joghurtbecher08115, but I finally have a test build if you're still interested in taking a look. Please fill out the form here: https://forms.gle/nwfZRSTzXpWa5vsH6 to get access.

Sounds great, thanks a lot, I just filled out the form :-) And no problem, since you just applied to be an Apple developer I don‘t think about one month is such a long time :-)

**Comment by retrogamer0815** [comment]
Hi, I have used it a few hours so far and it works quite well for a first release, filtering seems to work reliable so far on both my iPhone 8 and 2017 iPad Pro. Also exporting and importing of settings/filter lists between the devices worked via AirDrop or a cloud storage service :-)
One issue I ran into is when I tried to import the settings from Firefox on my desktop computer, on my iPhone and iPad I get an error message that the file is too large. The file I exported on my desktop computer is a little above 50 KB large and after entering all kinds of filter on my iPad manually I got the same error message when I got to about 16 KB. Is this a limit set by Apple for now, or could you allow a higher number of filters with a future update?
Oh and by the way, I made a report about this via Testflight with a screenshot too, would you prefer reports there or here?

Thanks again and have a great weekend!

**Reply by richardfrost** [reply]
Thanks so much for taking it for a spin and providing that feedback @Joghurtbecher08115! I can't find any published numbers on the limit, but it is the browser that sets the limit. I will test a couple things and see if I can expand it at all (possibly moving to local storage). Another thing I could do, and have thought about for a while, is optimizing the storage. Right now the config is pretty human friendly, but it doesn't have to be. It wouldn't be earth-shattering on the savings, but it could help support for a few more words. Just out of curiosity, do you recall about how many words that was when you hit 16KB?

As for reporting feedback like that, I'm good with whatever is more convenient for you. TestFlight and Github both work great, although back-and-forth communication might be easier in Github.

**Reply by retrogamer0815** [reply]
Alright, would be great, if there is a way to expand the size limit since depending on the use cases and native language 16 KB might not be enough. I checked the list I created on my iPhone 8 again and the size where I start getting the error message when trying to add another word is close to 17 KB and I have a little above 70 words in my list. That might be plenty for many people, however I have the problem, that in my native language there are for every single word up to six variants depending on the gender and grammatical cases and combinations of those two, so the list fills up quite quickly. In some cases you can save some slots for a few more words if you are smart about making the filters, but there is no generell solution unfortunately.

Otherwise I don't use Safari on my Mac as my main browser and there are plenty of options to use this extension in other browsers but I wouldn't mind doing some testing in Safari on my Mac too, however it seems that it can also be used in TestFlight right now, but only if you have a Mac with Apple Silicon chips. Are Mac apps for TestFlight really only supported on new Macs with Apple Silicon chips (which are only on the market for about a year by now), or do you maybe just need to change some settings to support Macs with Intel chips too?

![Screenshot 2021-11-15 at 14 37 30](https://user-images.githubusercontent.com/83663417/141996805-a40711c1-1c92-4df4-a8a3-d5418dba17be.png)

Otherwise I didn't run into any other issues yet and the extension so far always did its job and didn't suddenly stopped working.

**Reply by richardfrost** [reply]
Alright, I did some quick testing with a change to how data is stored and it let me successfully use config that was ~20KB, so I think that there is definitely a possibility to work around the limitation. I'll work on getting that, but it's a pretty big undertaking as it will effect the entire storage layer of the app. It should be more flexible after though, so that's cool!

As for the Silicon chip limitation, I think that's a limitation of TestFlight. It should be possible to use it on an Intel based Mac when its fully released, but I don't think I can enable it in this demo mode. I have tested it though, and it does work like you'd expect.

**Reply by retrogamer0815** [reply]
Thanks for having a look into this so quickly and letting me know, that is a huge relief :-) I think a limit of about 16 KB for such a config file would have been a problem for other extensions as well, for example in the Dark Reader extension I also use, you can choose on which sites the extension is changing the website to a dark design and on which not. So maybe not on day 1 but over time 16 KB would be a problem here for more and more people too.

One thing came to my mind when I used the extension today on my phone, in the settings window on the top right you have those buttons where people can make donations to you. I don't know the exact rules and if there are exceptions for donations but Apple famously only allows payments only over their own AppStore payment infrastructure (which is something, which is right now under investigation in various countries, so that might change). So far at least I did do donate a few times to developers from within apps for iPhone and iPad and I don't think there were other ways to do this than by doing it like an in-app-purchase, so it might be, that you may have to remove the buttons which are in the app, a least for now. But than again, I don't really know, but maybe something you could ask before submitting it to the AppStore ;-)

**Comment by retrogamer0815** [comment]
Hello and I think it is just not too late ti wish a Happy New Year :-)

I just noticed that by now in Testflight it says, that the test version of the app is now less than 30 days usable until I guess it needs to updated or at least re-uploaded by you. So it would be great if at least it could be re-uploaded in time, since I already feel much comfortable surfing the web on my mobile devices already with this pre-release version ;-)

Otherwise I didn't run into any big issues, one thing that was a little strange is that recently suddenly all my settings vanished on both devices (iPhone and iPad) and it was like the extension was just freshly installed. It might be connected to both of these devices being updated to iOS/iPadOS 15.2 which happened not long before that but I didn't check for or noticed this right after the updates and can't reproduce it now.

Are there otherwise any functions or features you think should be tested or you are interested in being tested? I use mainly test filters and some are more complex and work fine. What I also did a few times is importing and exporting where I ran into this issue with the size limit right now in iOS/iPadOS. What I didn't use yet ever and I personally don't feel any need for are the audio filters, but if you like and it helps, I could try this function on a few sites. I normally don't run into any bad words in videos since I almost never watch videos on social networks but I guess people who watch videos made public by private people a lot might have a need for that ;-)

**Reply by richardfrost** [reply]
Happy New Year! Sorry its been so slow.. I had a bunch of stuff come up that I had to take care of, and then I took a break for the holidays.

I'm so glad to hear that you are enjoying it though! I'm really excited to get it out there officially. 

I just published a new build that aims to address the storage issue of not being able to save more words. To activate it you'll need to disable the "Sync large settings" option (Options page > Config tab). One thing that would be helpful to test is if you can try switching that back and forth. If you have too many words it  and try to switch back to syncing it should throw an error and roll back the changes. If you have some time to test with it that would be great though. I want to feel more confident that it will work and not lose people's data. Another interesting thing to test would be downloading data. In the simulator it had trouble downloading the .json when I had too many words for the sync storage, but maybe it will work on a real device?

Sorry that you ended up losing your data.. I am not sure how that happened. Hopefully we can get it pretty solid before the real release.

As for any other things to test... I think the main thing I'd like to know if you've come across any sites it isn't working on? Any features that you usually work that have caused issues? If you don't use the audio muting, don't feel like you need to test it. I've tested it briefly on Netflix and YouTube with good success.

**Reply by retrogamer0815** [reply]
Thanks for the update, I just gave it a try and I could import my settings I exported from my desktop browser which until now were too big :-)

Here I first ran into an issue which I found a workaround for but probably should be made easier for other users later. I exported my settings from the extension on my desktop computer and saved it to a cloud drive. Then I went to the extension in Safari on my iPhone and under Config deactivated the setting "Sync large settings", which gave me a warning and seems to reset all settings. Next I imported the settings of my desktop browser from the cloud drive, however it first gave me the same error again saying, that the settings file is too big. What I noticed is, that trying to load the settings from the desktop browser seems to have activated the setting "Sync large settings" again. What I did next is look for that setting by opening a newly exported setting file from the phone in a text editor and I found the line  "syncLargeKeys": false, which I added to the settings file from my desktop browser and after that I could finally import them to APF under Safari on my phone and tablet :-) So I think somehow this setting under Config should maybe also be added to the desktop versions of the extension or if there is no value for it in an imported setting file it maybe shouldn't change the settings that are there at the moment which in my case was turned off when I first attempted to do this.

Otherwise after having imported the settings I tried on both devices a few times to turn on the option "Sync large settings" a few times, which first gives me the warning and after that the error message about the file size. The option stays turned off and the settings don't get reset by this anymore, so I think that works fine as it should, but I can go on doing this a few more times over the next days and weeks to be sure :-)

I didn't notice any negative side effects so far like websites complaining about this like they do sometimes about add-blockers and otherwise it worked reliable and I only noticed it not filtering anymore when suddenly all settings were gone which like I said might be related to an iOS update since it happened on both my phone and tablet around the same time and this is the only major thing happening on those devices I can think of which might have triggered it. And that wasn't a big deal since I just could import the settings again from a backup, so I didn't lose any data ;-)

No related to just the iOS/iPadOS versions, but since you asked about sites where it doesn't work: I had to deal with a PDF a few days ago which contained some wording I didn't like, so I checked for some ways to read it in a browser with this extension. Just using the inbuilt PDF-reader seems not to work, which I didn't expect but a first question I have is, would this theoretically be possible?
But for now I tried this site where you can upload a PDF to convert and view it in HTML. It looks not always great after doing that but if you just want to read the content that is enough:

https://cloudconvert.com/pdf-to-html#:~:text=CloudConvert%20is%20an%20online%20document,latest%20Microsoft%20Office%202019%20suite.

However I noticed that APF sometimes works on those converted files, sometimes it doesn't and sometimes just on the first few lines. The viewed tex can always be selected and copied by the mouse, so the conversion of it seems to have worked but still the filtering isn't reliable on those.

Nothing dramatic and those are probably just not normal cases and normal websites, but if something could be done about those, either by making the extension work on PDFs or more reliable on such converted files that would be a nice bonus, if you are looking for new ways to improve it ;-)

**Reply by richardfrost** [reply]
Thanks for the response! I'm sorry I forgot about mentioning the "Sync large settings". It is definitely being planned to add that to the desktop version, but I wanted to make sure it was working on iOS before doing that, just to avoid some migration nightmares if I had to change the method I was using to switch back and forth. That's heartening to hear its working for you once you switched that out.

I haven't seen it reset the settings, and don't think it should do that when switching back and forth. If you can re-produce that let me know what steps you took please.

I think I am going to add some more functionality on the import to automatically handle the storage options. I can't do it on export, because exporting from Chrome/Firefox desktop has different limits than mobile Safari, but before importing the data I could do a quick check to see how large the file/text is, and if its over what the current browser can handle I can automatically switch it to not sync the larger settings. That way it would import the config based on the current browser's limits. It will try to use the sync storage first (because that's the default), but if it can't it can then switch over. If the exported config is already set to not sync those large configs then it will just go ahead and put them in local storage. I'll see if I can get a build that does that out today.

As for PDFs, unfortunately there is usually not a lot that the filter can do with them, because even though they are text, the browser doesn't usually let extensions run on them. I also don't think its very feasible for the extension to do the conversion of PDFs, but I will add it to my list to take a look at the site you mentioned and see if I can figure out why the filter might not be working on that page. I'm definitely always on the lookout to try and find new ways to improve the application, so hopefully we'll find something that we can make better with that. I do have a project that is capable of filtering `.epub` files if that's something that you might find helpful. I haven't touched it in several months, but it uses the same config as the browser extension.

One last question that would really help me. Do things like the "Filter Method" (or anything besides words and domains) sync between devices? I don't have 2 devices to test it on, so I haven't been able to see if they actually sync those settings. You could also reset to defaults and then change something, like the filter method to "Censor" and then see if it also changes on the other device. I'm just curious if that works.

**Reply by retrogamer0815** [reply]
Hello, I set filters that replaced the English word "and" and "the" with "axd" and "txe" and did the same with the equivalents in my first language and did browse around with those for a while to see if run into a website, but so far the extension seems to work reliable, so if there are websites where it doesn't work, I would say that it probably isn't related to Safari on the iPhone or iPad. For now I removed those filters again since they got a little annoying when reading something but if there are some specific websites you would like me to have a look at just let me know.

When after upgrading the extension to version 0.1.0 (2) I turned off the option "Sync large settings" I am very positive that it reset the extension and I noticed that on both my iPhone and iPad, however it didn't bother me, since I wanted to import my settings from the desktop anyway. However I couldn't reproduce this so far, so maybe it had something to do with the update process before or something else. I did reset my extension myself again a few times and set some random filters, deactivated the setting and the newly set filters were still there. However I will keep an eye on my settings after future OS updates and from the extension.

I normally use the "Substitute" Method but I changed it on my desktop to "Remove" and exported/imported those settings to my phone and after that the method was also set to "Remove" on my phone. Also the switch to dark mode carried over.
I also compared some settings under "Match Method" which is sometimes different in my filters, I use "Exact" and "Partial" most of the time. As far as I can tell they also all carried over from my desktop to my phone and tablet, so that generally seems to work fine :-)

**Reply by richardfrost** [reply]
Thanks @Joghurtbecher08115 for all the great testing and feedback! I think that if its working that well for you, the core functionality seems to be pretty stable, which is great!

 I think I've wrapped up my storage updates, and am planning to push a new version tonight. There were a few pretty large changes to it, so I'd recommend grabbing a backup (if you don't already have one) and then import your config (you could even restore defaults for good measure).

**Reply by richardfrost** [reply]
@Joghurtbecher08115 I just published build 0.1.0 (5) which should be available now. Let me know if you have any questions about it.

I also (think) I added a build for MacOS that should work with both Apple/Intel architectures, build 0.1.0 (3). I don't know exactly how to access it, but its published in Test Flight as well, and you should have access. Let me know how that goes, if you want to test it there as well. I'm curious to know if syncing works (things like which filter method is active, for instance).

**Comment by retrogamer0815** [comment]
Hello,

I tested the new version for iOS/iPadOS over the weekend, on my iPad after the updated all my settings were still there, on my iPhone it did reset the extension, unfortunately I have no idea why, since the settings on both devices were identical as far as I can tell, but since I make backups regularly and the most up to date settings I normally have on my desktop computer this was no issue.
I reset the settings on my iPad after that too and tried to import a freshly exported settings file from my desktop machine where I didn't insert this line for the sync settings this time and I got a new message which told me to import those settings sync had to be turned off and I could chose to do it from there, which worked on both devices, so this issue I had before seems to be fixed now :-)

On my (Intel) Mac I could install the extension in Safari now too, the only thing that didn't work is the button on this message window, which should close the window and open the extensions preferences in the Safari settings, clicking on it doesn't do anything. However when I opened the extensions preferences myself the new extension was indeed there and I could activate it, so far it seems to work just fine under Safari on my Mac too.

![Screenshot](https://user-images.githubusercontent.com/83663417/150808340-921133f2-5704-45c5-811c-02d332c84d28.png)

**Reply by richardfrost** [reply]
@Joghurtbecher08115 Thanks for the very detailed feedback, its really helpful! That is great news that it seems to be working as expected in most counts. I'll take a look at that Safari button and see if there is anything I can do to improve that. Do you have another extension that works to confirm that button works on theirs? Maybe Dark Reader? I didn't really customize any of that part, it was just built-in when I packaged it for Intel Macs.

As for the config getting erased, I am not sure what could be causing that... Maybe it has something to do with the fact that it's a pre-release build? I'm just using the storage provided by Safari for extensions, so there isn't a lot I can do with it if its outside of the extension. Thanks for keeping an eye on that though, I'll definitely keep it in mind.

**Comment by retrogamer0815** [comment]
No problem, I already purchased Dark Reader for Safari on my phone and tablet and this already included the version for Safari on macOS, so I could install it there too. This comes with a small app as well which doesn't do much more than telling me, that the extension needs to be activated under the Safari settings and allows me to directly open those settings by clicking on a button.
For Dark Reader this button says Activate for Safari" which doesn't actually already activate the extension but clicking this button opens Safari and you are directly in the settings where you can activate newly installed extensions:

![Screenshot](https://user-images.githubusercontent.com/83663417/150934310-a6f95d50-fd36-4c91-a8c0-88d008109e87.png)

I am not very worried about the reset settings, since so far this are early pre-releases which don't come from the official store yet, so maybe there are some limitations. Also in one case I am sure that it happened after the OS update to iOS/iPadOS 5.2 and the last time you said yourself, that 0.1.0 (5) came with many changes, which I guess won't have to be done again once the extension is stable :-)

**Reply by richardfrost** [reply]
Thanks so much for sharing that. I'll take a look at it soon. I just released version [2.30.0](https://github.com/richardfrost/AdvancedProfanityFilter/releases/tag/2.30.0) for the main extension that includes the local storage option, just like the one you've been testing here. I just have a few smaller items left on the list before submitting a final version. Sadly I don't have a time estimate yet because work has been crazy, but I feel like its getting really close now.

**Reply by retrogamer0815** [reply]
No problem and take your time, just make sure you release at least an update before the latest test build expires ;-)

Otherwise Apple released version 15.3 for iOS and iPadOS the other day. Last time when 15.2 was released on both of my devices the settings got lost by the updates, this time the extension was not effected and I didn't have to import a backup after the updates :-)

**Reply by richardfrost** [reply]
I'll do my best ;-). I've got an overhaul for dark mode coming soon which will use the device's preference as the default.

That's great news that your settings stayed after the update.

**Reply by richardfrost** [reply]
@Joghurtbecher08115 I just published a new build for each (iOS/MacOS) that should be available now. Please see the notes on what's changed and for you to check up on them.

The latest MacOS one should have that button fixed in the app when it was supposed to open Safari, so let me know how that works for you if you get a chance. Thanks!

**Reply by retrogamer0815** [reply]
Hello, thanks, I just had a look on the new version on my Mac and phone, the button on the Mac works now and opens Safari and the section where you can turn on and off the extensions.
Switching between light and dark mode according to the system settings also worked on both devices once turning that setting on in the main settings of the app, so nothing to complain about the changes from my side :-)

One thing on the Mac you might have a look into is the text in the drop down menu when you click on the icon of the extension. The text in the selection menus on the right is maybe a little hard to read in both light mode and dark mode bot especially in dark mode:

**Edit:** I just noticed that this only happens when there is no page opened and the menus are inactive, in that case this is I guess supposed to look like this to indicate that they are not active, otherwise when a page is open they are fine readable ;-)

![Screenshot 2022-02-06 at 21 11 40](https://user-images.githubusercontent.com/83663417/152699806-e3fa3f56-740f-47ff-8db5-b5e47751bb11.png)

![Screenshot 2022-02-06 at 21 12 12](https://user-images.githubusercontent.com/83663417/152699808-41af3b42-ab54-4775-bc13-53cbb503090e.png)

**Comment by retrogamer0815** [comment]
Hi, since the other day you released another preview release for Apple devices I installed them now on one device after another. The new release still seems to work just fine and I didn't notice any issues. Just again on one device (the iPad) after the update the extension behaved like it was installed the first time, I was asked to give it the necessary permissions and all settings and word lists were gone. Again not a huge problem for me since I regularly make backups of my lists but might be a bigger issue of course for people who don't. Also I still have no idea why this happened, the settings are identical and all my devices are always up to date on the lates version of iOS/iPadOS.

**Reply by richardfrost** [reply]
Thanks for the continued reports on how its working for you. I'm pretty baffled by it as well. I haven't noticed the same behavior on my iPad. I am really hoping it might just be some weirdness related to how the TestFlight system works, and hope that it won't affect things once added to the App Store.

I hope to have a final build ready for Apple reviewers posted this week, so I think we should be getting really close! Feel free to reach out with any additional info or observations if you come across them. Thanks!

**Reply by retrogamer0815** [reply]
Alright and no problem, since it happens to me on just one device and never on your side, I agree that it is probably just some rare or weird circumstance with Testflight or something on my iPad, which is an older 2017 model which by now has a long OS update history, maybe this problem would go away if I just did a clean reinstall of iPadOS.
Looking forward to the final release, I will keep you updated about how that works in the end and if I run into any issues.

**Comment by richardfrost** [comment]
I'm excited to announce that Apple has approved the APF app, and it is available for purchase in the App Store here: https://apps.apple.com/us/app/advanced-profanity-filter/id1593810102

It is available for both iOS (15+) and macOS. Due to the cost of being an Apple developer, I won't be able share it for free at this time, so it will require a one-time purchase. The purchase will unlock it for iOS and macOS and can be shared with everyone in your "Apple Family".

Thanks for helping with the testing and all the valuable feedback @Joghurtbecher08115

**Reply by Lhickox1** [reply]
Any way to add the site 9anime.se for audio muting?

**Reply by retrogamer0815** [reply]
> I'm excited to announce that Apple has approved the APF app, and it is available for purchase in the App Store here: https://apps.apple.com/us/app/advanced-profanity-filter/id1593810102
> 
> It is available for both iOS (15+) and macOS. Due to the cost of being an Apple developer, I won't be able share it for free at this time, so it will require a one-time purchase. The purchase will unlock it for iOS and macOS and can be shared with everyone in your "Apple Family".
> 
> Thanks for helping with the testing and all the valuable feedback @Joghurtbecher08115

Finally :-) I just purchased it and updates so far on my iPad and iPhone from the Testflight version to the official one worked fine without any issues. The price is reasonable I think since the app isn’t add supported and being a registered Apple developer isn’t free, so I hope most people will agree, I will add another donation to it later today or over the weekend ;-)

**Reply by richardfrost** [reply]
Thanks @Joghurtbecher08115, I'm so glad its finally available. Feel free to reach out if you have any trouble with it, or any other feedback. Also, if you get some time, I would really appreciate a review since it's a new app on the App Store.

**Reply by Lhickox1** [reply]
I have been working on adding the site 9anime.se using your instructions on how to add sites, but after hours of trial and error, I still can't add it. The issue may be because you can only inspect the site with a video open on foxfire and because of my limited experience. It would help to have some updated instructions about how to add sites on foxfire.

**Reply by richardfrost** [reply]
Did you have a chance to try what I added in issue #402 @Lhickox1?

**Reply by retrogamer0815** [reply]
> Thanks @Joghurtbecher08115, I'm so glad its finally available. Feel free to reach out if you have any trouble with it, or any other feedback. Also, if you get some time, I would really appreciate a review since it's a new app on the App Store.

Hi, using the final version now more than a month I can say, that so far in my case I didn't have any issue :-) And of course I gave it a positive review in the AppStore and convinced some friends that this extension is absolutely necessary to live a happy life ;-)

Otherwise two things:

First just these days there was again the WWDC and even though I am not a developer I enjoy watching some of the videos every years since it still is interesting to me when they explain some basic technologies. I noticed that there is a session where they explain new things for Safari extension coming with the iOS and macOS updates later this year, maybe this could be of interest to you, they also talk about storage:

https://developer.apple.com/videos/play/wwdc2022-10099

Otherwise I have some feature suggestions I have been thinking about for a while, but I wanted to wait until the Safari ports are done ;-)

1. In my native language there are cases where it can matter when at the beginning or in a word a letter is a capital letter or not, however right now in APF there is only an option for the replacements to make it case sensitive. So I wonder if an option could be added to set a filter where the extension searches for "dogHouse" but not "doghouse"?


2. Also in case of grammar my native language is unfortunately (and in my opinion unnecessary) complicated and words change more often that in English (where I think there aren't many changes beside singular and plural). What I could use would be an additional option for a filter to check the word right before the one I wan't to filter and depending on it, choose the replacement.
I try to explain it by an example in my language (German) where I replace Birnen (pears) by Äpfel (apples).

I could first simply create a rule where I let the extension look for "Birnen" and "Äpfel" as the replacement. However this way I don't always get the right replacement since when used in the accusative case the plural form of "Apfel" is "Äpfeln" with an additional N, however the plural form of "Birne" is always the same.

So i goes like this:

singular: 1 Apfel,  plural: 2 Äpfel,  accusative case: den Äpfeln

singular: 1 Birne, plural: 2 Birnen, accusative case: den Birnen

So since I can't differentiate "Birnen" which is the same in all plural cases I can't right now make a filter that replaces it with "Äpfel" or "Äpfeln" depending on what case is used. However in most cases you this could be solved, if the word before is checked.

So do you think it would be possible sometimes in the future to add a option that would work like this:

Look for "Birnen", if word before "Birnen" is "den" replace with "Äpfeln", if not replace with "Äpfel"?

Right now of course I can make a rule for every possible combination I can imagine right now, but it makes my list quite long and I have some wildcard rules that would make many of those rules unnecessary  ;-)


3. Also I was wondering if an integration of a cloud provider like Dropbox, OneDrive etc. might be possible for syncing the extension on different browsers, since I use it on five machines by now. I kind of keep them in sync by keeping the latest filter list on a cloud drive, but I have to export/import it from time to time on all machines, having this happen automatically would make this easier. This one as the last suggestion since for me is the lowest priority, just something the would be nice to have and if you wonder how to improve the extension next ;-)
